<launch>
  <arg name="build_only" default="false" description="shutdown node after TensorRT engine file is built"/>
  <arg name="model_path" default="$(env HOME)/autoware_data/camera_streampetr"/>

  <arg name="model_param_path" default="$(var model_path)/ml_package_camera_streampetr.param.yaml"/>
  <arg name="node_param_path" default="$(find-pkg-share autoware_camera_streampetr)/config/camera_streampetr.param.yaml"/>

  <arg name="is_compressed_image" default="false" description="whether the input image is compressed"/>

  <let name="compressed_suffix" value="$(eval &quot;'/compressed' if '$(var is_compressed_image)' == 'true' else ''&quot;)"/>
  <let name="camera0_topic" value="/sensing/camera/camera0/image_rect_color$(var compressed_suffix)"/>
  <let name="camera1_topic" value="/sensing/camera/camera1/image_rect_color$(var compressed_suffix)"/>
  <let name="camera2_topic" value="/sensing/camera/camera2/image_rect_color$(var compressed_suffix)"/>
  <let name="camera3_topic" value="/sensing/camera/camera3/image_rect_color$(var compressed_suffix)"/>
  <let name="camera4_topic" value="/sensing/camera/camera4/image_rect_color$(var compressed_suffix)"/>

  <node pkg="autoware_camera_streampetr" exec="autoware_camera_streampetr_node" name="stream_petr" output="screen">
    <param from="$(var model_param_path)" allow_substs="true"/>
    <param from="$(var node_param_path)" allow_substs="true"/>
    <param name="build_only" value="$(var build_only)"/>

    <!-- Image of ~/input/camera_i will be the ith image in the model input -->
    <remap from="~/input/camera0/camera_info" to="/sensing/camera/camera0/camera_info"/>
    <remap from="~/input/camera1/camera_info" to="/sensing/camera/camera1/camera_info"/>
    <remap from="~/input/camera2/camera_info" to="/sensing/camera/camera2/camera_info"/>
    <remap from="~/input/camera3/camera_info" to="/sensing/camera/camera3/camera_info"/>
    <remap from="~/input/camera4/camera_info" to="/sensing/camera/camera4/camera_info"/>

    <remap from="~/input/camera0/image$(var compressed_suffix)" to="$(var camera0_topic)"/>
    <remap from="~/input/camera1/image$(var compressed_suffix)" to="$(var camera1_topic)"/>
    <remap from="~/input/camera2/image$(var compressed_suffix)" to="$(var camera2_topic)"/>
    <remap from="~/input/camera3/image$(var compressed_suffix)" to="$(var camera3_topic)"/>
    <remap from="~/input/camera4/image$(var compressed_suffix)" to="$(var camera4_topic)"/>
  </node>
</launch>
